src/components/RadioButtonGroup/RadioButtonGroup.tsx

```diff
@@ -40,34 +47,61 @@ export interface RadioButtonGroupProps extends HTMLProps<HTMLDivElement> {
    * The ref to the HTML Dom element
    */
   ref?: Ref<HTMLDivElement>;
+  /**
+   * Warning or error message, displayed in a tooltip.
+   */
+  validationHint?: string;
+  /**
+   * Triggers error styles on the component. Important for accessibility.
+   */
+  invalid?: boolean;
 }
 
+const inlineMessageStyles = ({ theme }: StyleProps) => css`
+  margin-top: ${theme.spacings.mega};
+`;
+
+const StyledInlineMessage = styled(InlineMessage)<InlineMessageProps>(
+  inlineMessageStyles,
+);
+
 const RadioButtonGroupComponent = (
   {
     options,
     onChange,
     value: activeValue,
     name: customName,
     label,
+    invalid,
+    validationHint,
     ...props
   }: RadioButtonGroupProps,
   ref: RadioButtonGroupProps['ref'],
 ) => {
   const name = customName || uniqueId('radio-button-group_');
   return (
-    <div role="group" aria-label={label} ref={ref} {...props}>
-      {options &&
-        options.map(({ children, value, className, ...rest }) => (
-          <div key={value && value.toString()} className={className}>
-            <RadioButton
-              {...{ ...rest, value, name, onChange }}
-              checked={value === activeValue}
-            >
-              {children}
-            </RadioButton>
-          </div>
-        ))}
-    </div>
+    <Label htmlFor={name}>
+      {label && <span>{label}</span>}
+
+      <div role="group" aria-label={label} ref={ref} {...props}>
+        {options &&
+          options.map(({ children, value, className, ...rest }) => (
+            <div key={value && value.toString()} className={className}>
+              <RadioButton
+                {...{ ...rest, value, name, onChange }}
+                checked={value === activeValue}
+              >
+                {children}
+              </RadioButton>
+            </div>
+          ))}
+      </div>
+      {invalid && (
+        <StyledInlineMessage variant="danger">
+          {validationHint}
+        </StyledInlineMessage>
+      )}
```

Comment:
Once you've replaced the InlineMessage component with the ValidationHint component, you could also add support for the `hasWarning` and `showValid` props that are supported by other input components.